client/my-sites/domains/domain-management/dns/dns-records-list.jsx

```diff
@@ -0,0 +1,267 @@
+		icon: (
+			<Icon
+				icon={ redo }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( 'Enable' ),
+		callback: ( record ) => this.addDns( record ),
+	};
+
+	recordInfoAction = {
+		icon: (
+			<Icon
+				icon={ info }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( "What's this?" ),
+		// TODO: Add the correct callback to show the info about Domain Connect record
+		callback: () => {},
+	};
+
+	editRecordAction = {
+		icon: (
+			<Icon
+				icon={ edit }
+				size={ 18 }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( 'Edit' ),
+		// TODO: Add the correct callback function once the DNS add page is complete
+		callback: () => {},
+	};
+
+	deleteRecordAction = {
+		icon: (
+			<Icon
+				icon={ trash }
+				size={ 18 }
+				className="gridicon dns-records-list__action-menu-item"
+				viewBox="2 2 20 20"
+			/>
+		),
+		title: this.props.translate( 'Delete' ),
+		callback: ( record ) => this.deleteDns( record ),
+	};
+
+	noDialog() {
+		return {
+			type: null,
+			onClose: null,
+		};
+	}
+
+	openDialog( type, onClose ) {
+		this.setState( {
+			dialog: {
+				type,
+				onClose,
+			},
+		} );
+	}
+
+	handleDialogClose = ( result ) => {
+		this.state.dialog.onClose( result );
+		this.setState( { dialog: this.noDialog() } );
+	};
+
+	deleteDns = ( record, action = 'delete', confirmed = false ) => {
+		const { selectedDomainName, translate } = this.props;
+		const { records } = this.props.dns;
+
+		if ( ! confirmed && isDeletingLastMXRecord( record, records ) ) {
+			this.openDialog( 'deleteEmailForwards', ( result ) => {
+				if ( result.shouldDeleteEmailForwards ) {
+					this.deleteDns( record, action, true );
+				}
+			} );
+
+			return;
+		}
+
+		const successMessage =
+			'delete' === action
+				? translate( 'The DNS record has been deleted.' )
+				: translate( 'The DNS record has been disabled.' );
+
+		const errorMessage =
+			'delete' === action
+				? translate( 'The DNS record has not been deleted.' )
+				: translate( 'The DNS record has not been disabled.' );
+
+		this.props.deleteDns( selectedDomainName, record ).then(
+			() => {
+				const successNoticeId = 'dns-list-success-notice';
+				this.props.successNotice( successMessage, {
+					id: successNoticeId,
+					showDismiss: false,
+					duration: 5000,
+					button: translate( 'Undo' ),
+					onClick: () => {
+						this.props.removeNotice( successNoticeId );
+						this.addDns( record );
+					},
+				} );
+			},
+			( error ) => {
+				this.props.errorNotice( error.message || errorMessage );
+			}
+		);
+	};
+
+	addDns( record ) {
+		const { translate } = this.props;
+
+		this.props.addDns( this.props.selectedDomainName, record ).then(
+			() => {
+				this.props.successNotice( translate( 'The DNS record has been enabled.' ), {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'The DNS record has been deleted.' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/dns-records-page-design)